src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -192,7 +192,100 @@ private DenseMatrix initRhs(LfNetwork lfNetwork, EquationSystem equationSystem,
+     * @param network
+     * @param lfNetwork
+     * @param oldFactors
+     * @return
+     */
+    private List<SensitivityFactor> getDistributedFactors(Network network, LfNetwork lfNetwork,
+                                                          List<SensitivityFactor> oldFactors) {
+        // todo: load ?
+        // todo: use only lfnetwork ?
+
+        List<SensitivityFactor> factors = new ArrayList<>();
+
+        Set<String> monitoredLines = new HashSet<>();
+        for (SensitivityFactor oldFactor : oldFactors) {
+            // if the factor if an injection then we will compute it with distribution, otherwise we just keep it as it is
+            if (oldFactor instanceof BranchFlowPerInjectionIncrease) {
+                BranchFlowPerInjectionIncrease branchFactor = (BranchFlowPerInjectionIncrease) oldFactor;
+                monitoredLines.add(branchFactor.getFunction().getBranchId());
+            } else {
+                factors.add(oldFactor);
+            }
+        }
+
+        monitoredLines.forEach(
+            lineId -> {
+                // todo: check that the line is still in the connected component ?
+                BranchFlow branchFlow = new BranchFlow(lineId, lineId, lineId);
+                network.getGenerators().forEach(// todo: not all generators, only those who have an impact in compensation
+                    generator -> {
+                        if (generator.getTerminal().getBusView().getBus() == null) {
+                            return;
+                        }
+                        // check connected component
+                        if (lfNetwork.getBusById(generator.getTerminal().getBusView().getBus().getId()) == null) {
+                            return;
+                        }
+                        String genId = generator.getId();
+                        factors.add(new BranchFlowPerInjectionIncrease(branchFlow,
+                                new InjectionIncrease(genId, genId, genId)));
+                    }
+                );
+            }
+        );
+        return factors;
+    }
+
+    /**
+     * Return a list of id from participating element, and its participation value
+     * @param lfNetwork
+     * @param loadFlowParameters
+     * @param openLoadFlowParameters
+     * @return
+     */
+    private List<Pair<String, Number>> getParticipatingElements(LfNetwork lfNetwork, LoadFlowParameters loadFlowParameters, OpenLoadFlowParameters openLoadFlowParameters) {
+        List<Pair<String, Number>> contributionList = new ArrayList<>();
+
+        if (loadFlowParameters.isDistributedSlack()) {
+            switch (loadFlowParameters.getBalanceType()) {
+                case PROPORTIONAL_TO_GENERATION_P_MAX:
+                    DistributedSlackOnGenerationOuterLoop outerLoop = new DistributedSlackOnGenerationOuterLoop(openLoadFlowParameters.isThrowsExceptionInCaseOfSlackDistributionFailure());
```

Comment:
This is were I need all this outer loop stuff to be Public, I would love to avoid this